### PR TITLE
Feature 39/manage user profile

### PR DESCRIPTION
**Pull request by akononenko2375**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/store/authorization/actions.ts [hunk 1]
```diff
@@ -35,20 +39,46 @@ export const actions: ActionTree<AuthorizationStateInterface, RootStateInterface
...
+    const userId = state.user!.id;
+    await HTTP.put(`/users/${userId}`, payloadData).then((r: AxiosResponse<UserInterface>) => {
+      commit(AuthMutationEnum.SET_USER, r.data);
+    });
+  },
+
+  [AuthActionEnum.DELETE_CONTACT]: async ({ state, commit, dispatch }, payload) => {
+    const userId = state.user!.id;
+    await HTTP.delete(`/users/${userId}/contacts/${payload}`);
+    await HTTP.get(`/users/${userId}/contacts`).then((r: AxiosResponse<ContactInterface>) => {
```
Comment (aleonovdementiev) [inline src/store/authorization/actions.ts]: Why do you need two requests here?

File: src/store/authorization/actions.ts [hunk 2]
```diff
@@ -35,20 +39,46 @@ export const actions: ActionTree<AuthorizationStateInterface, RootStateInterface
...
 
-    commit(AuthMutationEnum.SET_USER, null);
-  },
   [AuthActionEnum.SIGN_OUT]: ({ commit, dispatch }, payload) => {
     dispatch('localStorageStore/REMOVE', 'user', { root: true });
     commit(AuthMutationEnum.SET_USER, payload);
   },
+  /////////// MY ACTIONS
+  [AuthActionEnum.SET_FORM]: ({ commit }, payload) => {
+    commit(AuthMutationEnum.SET_FORM, payload);
```
Comment (aleonovdementiev) [inline src/store/authorization/actions.ts]: Why do you need this action? You can call mutation directly from component.

**Comment by aleonovdementiev** [comment]
Please, revert all changes in package-lock.json if you haven't changed anything in project packages dependencies

File: src/store/authorization/actions.ts (hunk 2 above)
Comment (akononenko2375) [reply src/store/authorization/actions.ts]: I agree. Already changed it.

File: src/store/authorization/actions.ts (hunk 1 above)
Comment (akononenko2375) [reply src/store/authorization/actions.ts]: The second request needed to update the contact in store. Without this request, I could not update the state.user.contacts.

Comment (aleonovdementiev) [reply src/store/authorization/actions.ts]: You can update contact in store in mutation with response payload, that you will get on delete action from BE

**Review by rostovskiy88 — DISMISSED** [review]

File: src/store/authorization/actions.ts [hunk 3]
```diff
@@ -35,20 +39,45 @@ export const actions: ActionTree<AuthorizationStateInterface, RootStateInterface
...
+    const userId = state.user!.id;
+    const payloadData = new PostContactModel(payload[0]);
+    await HTTP.post(`/users/${userId}/contacts`, payloadData)
+      .then((r) => {
+        commit(AuthMutationEnum.ADD_CONTACT, r.data);
+      })
+      .catch((e) => {
+        console.log(e.response.data.error_message);
+      });
+  },
```
Comment (aleonovdementiev) [inline src/store/authorization/actions.ts:81]: Please, use one style of error handling in all actions

Comment (akononenko2375) [reply src/store/authorization/actions.ts:81]: ok, done!

**Review by rostovskiy88 — APPROVED** [review]

**Review by aleonovdementiev — APPROVED** [review]

**Pull request merged**